src/repos/FlyerRepo.ts

```diff
@@ -0,0 +1,229 @@
+  return FlyerModel.find({ 'organization.slug': organization.slug })
+    .sort({ date: 'desc' })
+    .skip(offset)
+    .limit(limit)
+    .then((flyers) => {
+      return flyers.filter((flyer) => !isFlyerFiltered(flyer));
+    });
+};
+
+const getFlyersByOrganizationIDs = async (
+  organizationIDs: string[],
+  limit: number = DEFAULT_LIMIT,
+  offset: number = DEFAULT_OFFSET,
+): Promise<Flyer[]> => {
+  const uniqueOrgIDs = [...new Set(organizationIDs)].map((id) => new ObjectId(id));
+  console.log(uniqueOrgIDs);
+  const orgSlugs = await OrganizationModel.find({ _id: { $in: uniqueOrgIDs } }).select('slug');
+  return getFlyersByOrganizationSlugs(
+    orgSlugs.map((org) => org.slug),
+    limit,
+    offset,
+  );
+};
+
+const getFlyersAfterDate = async (since: string, limit = DEFAULT_LIMIT): Promise<Flyer[]> => {
+  return (
+    FlyerModel.find({
+      // Get all Flyers after or on the desired date
+      date: { $gte: new Date(new Date(since).setHours(0, 0, 0)) },
+    })
+      // Sort dates in order of most recent to least
+      .sort({ date: 'desc' })
+      .limit(limit)
+      .then((flyers) => {
+        return flyers.filter((flyer) => !isFlyerFiltered(flyer));
+      })
+  );
+};
+
+/**
+ * Performs fuzzy search on all Flyers to find Flyers with title/publisher matching the query.
+ * @param query the term to search for
+ * @param limit the number of results to return
+ * @returns at most limit Flyers with titles or publishers matching the query
+ */
+const searchFlyers = async (query: string, limit = DEFAULT_LIMIT) => {
+  const allFlyers = await FlyerModel.find({});
+  const searcher = new Fuse(allFlyers, {
+    keys: ['title', 'organization.name'],
+  });
+
+  return searcher
+    .search(query)
+    .map((searchRes) => searchRes.item)
+    .slice(0, limit);
+};
+
+/**
+ * Computes and returns the trending Flyers in the database.
+ *
+ * @function
+ * @param {number} limit - number of Flyers to retrieve.
+ */
+const getTrendingFlyers = async (limit = DEFAULT_LIMIT): Promise<Flyer[]> => {
+  const flyers = await FlyerModel.find({ isTrending: true }).exec();
+  return flyers.filter((flyer) => !isFlyerFiltered(flyer)).slice(0, limit);
+};
+
+/**
+ * Refreshes trending Flyers.
+ */
+const refreshTrendingFlyers = async (): Promise<Flyer[]> => {
+  // Set previous trending Flyers to not trending
+  const oldTrendingFlyers = await FlyerModel.find({ isTrending: true }).exec();
+  oldTrendingFlyers.forEach(async (a) => {
+    const flyer = await FlyerModel.findById(new ObjectId(a._id)); // eslint-disable-line
+    flyer.isTrending = false;
+    await flyer.save();
+  });
+
+  // Get new trending Flyers
+  const flyers = await FlyerModel.aggregate()
+    // Get a sample of random Flyers
+    .sample(100)
+    // Get Flyers after 30 days ago
+    .match({
+      date: {
+        $gte: new Date(
+          new Date().setDate(new Date().getDate() - MAX_NUM_DAYS_OF_TRENDING_ARTICLES),
+        ),
+      },
```

Comment:
and this
from @katesliang:
> how exactly are we determining if something is trending?
>
> from what i see we just get a random sample of 100 flyers and then see if they are from within the last 30 days?

Reply:
rn we're just randomly sampling because we don't have any flyers T_T but I can fix it later after we get the app launched

Reply:
okie you should leave a comment or create an issue or smth to document it somewhere

Reply:
let's create an issue! @isaachan100

Reply:
prod